src/lib/y2network/widgets/s390_common.rb

```diff
@@ -0,0 +1,322 @@
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        _("Options")
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = @settings.attributes
+      end
+
+      # @see CWM::AbstractWidget
+      def opt
+        [:hstretch]
+      end
+
+      # @see CWM::AbstractWidget
+      def help
+        # TRANSLATORS: S/390 dialog help for QETH Options
+        _("<p>Enter any additional <b>Options</b> for this interface (separated by spaces).</p>")
+      end
+
+      # @see CWM::AbstractWidget
+      def store
+        @settings.attributes = value
+      end
+    end
+
+    class S390IPAddressTakeover < CWM::CheckBox
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = !!@settings.ipa_takeover
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        _("Enable IPA takeover")
+      end
+
+      # @see CWM::AbstractWidget
+      def help
+        _("<p>Select <b>Enable IPA Takeover</b> if IP address takeover should be enabled " \
+          "for this interface.</p>")
+      end
+
+      # @see CWM::AbstractWidget
+      def store
+        @settings.ipa_takeover = value
+      end
+    end
+
+    class S390Layer2 < CWM::CustomWidget
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+        @settings = settings
+        self.handle_all_events = true
+      end
+
+      def contents
+        VBox(
+          Left(support_widget),
+          Left(mac_address_widget)
+        )
+      end
+
+      def init
+        refresh
+      end
+
+      def handle(event)
+        case event["ID"]
+        when support_widget.widget_id, mac_address_widget.widget_id
+          refresh
+        end
+
+        nil
+      end
+
+    private
+
+      def refresh
+        support_widget.checked? ? mac_address_widget.enable : mac_address_widget.disable
+      end
+
+      def support_widget
+        @support_widget ||= S390Layer2Support.new(@settings)
+      end
+
+      def mac_address_widget
+        @mac_address_widget ||= S390Layer2Address.new(@settings)
+      end
+    end
+
+    class S390Layer2Support < CWM::CheckBox
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = !!@settings.layer2
+      end
+
+      # @see CWM::AbstractWidget
+      def opt
+        [:notify]
```

Comment:
why notify when you do not have handle?

Reply:
Well, it is not handled by the widget itself but by the CustomWIdget that cointains layer2 support and mac widgets.

https://github.com/yast/yast-network/pull/911/files/c2fd9b89ca1b0072ca7cdc305ed99cdad5917723#diff-3cbb037cd3044597452d2cdbe8e7f71aR227

Reply:
so please document it, as it is not obvious